File: Python3/프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    N, M = len(land), 4
    answer = 0
    
    tmp = 0
    for i in range(1, N):
        for j in range(M):
            tmp = 0
            for k in range(M):
                if j == k:
                    continue
                tmp = max(tmp, land[i][j] + land[i - 1][k])
            land[i][j] = tmp
                
    return max(land[N - 1])

File: C++/프로그래머스/2/12913. 땅따먹기/땅따먹기.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> land) {
    int N = land.size(), M = 4;
    
    int tmp;
    for(int i = 1; i < N; i++) {
        for(int j = 0; j < M; j++) {
            tmp = 0;
            for(int k = 0; k < M; k++) {
                if(j == k) continue;
                tmp = max(tmp, land[i][j] + land[i - 1][k]);
            }
            land[i][j] = tmp;
        }
    }

    return *max_element(land[N - 1].begin(), land[N - 1].end());
}